File: single-number-III/singleNumber3.go
func singleNumber(nums []int) []int {
    var diff int = 0
    var a,b int = 0,0
    for _,i := range nums {
        diff ^= i
    }
    diff &= -diff
    for _,i := range nums {
        if i & diff == 0 {
            a ^= i
        }else {
            b ^= i
        }
    }
    return []int{a,b}
}

File: single-number-III/singleNumber3.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: List[int]
        """
        diff = reduce(lambda x, y: x ^ y, nums)
        diff &= -diff
        return [reduce(lambda x, y: x ^ y, [i for i in nums if i & diff == 0]), reduce(lambda x, y: x ^ y, [i for i in nums if i & diff != 0])]
